### DISCUSSION TITLE
Installing 3.12-3.13 on Ubuntu: Some packages could not be installed

### DISCUSSION THREAD

**Discussion by bekoev** [rabbitmq/rabbitmq-server#12924] [Questions] [answered]
### Describe the bug

Following https://www.rabbitmq.com/docs/install-debian#apt-quick-start-cloudsmith on Ubuntu 22.04 with the following change
```
sudo apt install rabbitmq-server=3.13.7-1 --fix-missing
```
leads to the error:
```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 rabbitmq-server : Depends: erlang-base (< 1:27.0) but 1:27.2-1 is to be installed or
                            erlang-base-hipe (< 1:27.0) but it is not installable or
                            esl-erlang (< 1:27.0) but it is not installable
                   Depends: erlang-crypto (< 1:27.0) but 1:27.2-1 is to be installed or
...
```

### Reproduction steps

1. Ubuntu 22.04
2. Follow all the steps, except the last, from https://www.rabbitmq.com/docs/install-debian#apt-quick-start-cloudsmith
3. `sudo apt install rabbitmq-server=3.13.7-1 --fix-missing`

### Expected behavior

```
Preparing to unpack .../rabbitmq-server_3.13.7-1_all.deb ...
Unpacking rabbitmq-server (3.13.7-1) ...
Setting up rabbitmq-server (3.13.7-1) ...
Adding group `rabbitmq' (GID 119) ...
Done.
```

### Additional context

Changing the step
```
## Install Erlang packages
sudo apt-get install -y erlang-base \
                        erlang-asn1 erlang-crypto erlang-eldap erlang-ftp erlang-inets \
                        erlang-mnesia erlang-os-mon erlang-parsetools erlang-public-key \
                        erlang-runtime-tools erlang-snmp erlang-ssl \
                        erlang-syntax-tools erlang-tftp erlang-tools erlang-xmerl
```
to
```
sudo apt-get install -y erlang-base=1:26.2.5.6-1 \
                        erlang-asn1=1:26.2.5.6-1 erlang-crypto=1:26.2.5.6-1 erlang-eldap=1:26.2.5.6-1 erlang-ftp=1:26.2.5.6-1 erlang-inets=1:26.2.5.6-1 \
                        erlang-mnesia=1:26.2.5.6-1 erlang-os-mon=1:26.2.5.6-1 erlang-parsetools=1:26.2.5.6-1 erlang-public-key=1:26.2.5.6-1 \
                        erlang-runtime-tools=1:26.2.5.6-1 erlang-snmp=1:26.2.5.6-1 erlang-ssl=1:26.2.5.6-1 \
                        erlang-syntax-tools=1:26.2.5.6-1 erlang-tftp=1:26.2.5.6-1 erlang-tools=1:26.2.5.6-1 erlang-xmerl=1:26.2.5.6-1
```
seems to help.

**Comment by lukebakken** [comment, marked as answer]
Thanks for using RabbitMQ.

When you install Erlang according to our docs, it installs the version of Erlang that the latest version of RabbitMQ (4.x) requires.

Basically, you're installing the wrong version of Erlang for RabbitMQ 3.13, and expecting the package manager to fix it for you, which, as you've found, doesn't work.

**Reply by bekoev** [reply]
Hi Luke, thank you for your response. 
Could you please also advise which of the following is more correct? 
1. The RabbitMQ team does not have any possibility to support the package manager in successful automatic resolution, e.g. by configuring the repositories maintained by the team. 
2. The RabbitMQ team does not have resources for that.

**Reply by michaelklishin** [reply]
@bekoev with `apt`, if you do not specify a version to install, the most recent available one will be installed. A dedicated doc section that explains [how to pin package versions with `apt`](https://www.rabbitmq.com/docs/install-debian#apt-pinning) has been around for years.

The docs target RabbitMQ `4.0.x` which does support Erlang 27. We can add an explicit version to the `3.13.x` version of the site, or so can anyone else since [the website is open source](https://github.com/rabbitmq/rabbitmq-website).

**Reply by michaelklishin** [reply]
A mandatory mention of the fact that both `3.12.x` and `3.13.x` are [out of community support](https://www.rabbitmq.com/release-information). The docs can be adapted for the Erlang 27.x era but don't expect any support for series other than `4.0.x` unless you have a commercial support license.

**Reply by bekoev** [reply]
Thanks a lot, Michael!

Let's see if I can reduce the amount of future similar requests by https://github.com/rabbitmq/rabbitmq-website/pull/2141
